tests/integration/test_charm.py

```diff
@@ -90,37 +99,54 @@ async def test_metrics_endpoint(ops_test: OpsTest):
     assert mimir_targets
 
 
-async def test_mimir_cluster(ops_test: OpsTest):
+async def test_mimir_monolithic(ops_test: OpsTest):
     assert ops_test.model is not None
     await ops_test.model.deploy(
         "mimir-worker-k8s",
         "worker",
         channel="latest/edge",
         config={"role-all": True, "role-query-frontend": True},
     )
-    await ops_test.model.deploy("grafana-agent-k8s", "agent")
-
     await ops_test.model.integrate("mimir:mimir-cluster", "worker")
-    await ops_test.model.integrate("grafana:metrics-endpoint", "agent")
-    await ops_test.model.integrate("mimir:receive-remote-write", "agent")
-    await ops_test.model.integrate("mimir:s3", "s3")
-
     await ops_test.model.wait_for_idle(apps=["mimir", "worker", "agent", "s3"], status="active")
+    await check_agent_data_in_mimir(ops_test, "mimir")
+    await ops_test.model.remove_application(app_name="worker", destroy_storage=True)
```

Comment:
I'm not very comfortable with doing model teardown stuff inside a test function. 
Try to follow this pattern https://discourse.charmhub.io/t/classify-tests-with-pytest-custom-markers-for-quick-integration-testing-iterations/14006

it's cleaner, and allows for much faster iterations when writing the tests

Reply:
I'm splitting the tests, because all the workers using the same s3 backend isn't really testing much, as the data would be saved in one place only.